sarracen/interpolate.py

```diff
@@ -302,3 +302,120 @@ def _fast_interpolate3d_cross(xterm, yterm, zterm, wfunc, zslice, kernrad, targe
             np.sqrt(q2), 3)
 
     return image
+
+
+def interpolate3D(data: 'SarracenDataFrame',
+                  x: str,
+                  y: str,
+                  target: str,
+                  kernel: BaseKernel,
+                  pixwidthx: float,
+                  pixwidthy: float,
+                  xmin: float = 0,
+                  ymin: float = 0,
+                  pixcountx: int = 480,
+                  pixcounty: int = 480,
+                  int_samples: int = 1000):
+    """
+    Interpolates particle data in a SarracenDataFrame across three directional axes to a 2D
+    grid of pixels, by summing contributions in columns across the z-axis.
+
+    :param data: The particle data, in a SarracenDataFrame.
+    :param x: The column label of the x-directional axis.
+    :param y: The column label of the y-directional axis.
+    :param target: The column label of the target smoothing data.
+    :param kernel: The kernel to use for smoothing the target data.
+    :param pixwidthx: The width that each pixel represents in particle data space.
+    :param pixwidthy: The height that each pixel represents in particle data space.
+    :param xmin: The starting x-coordinate (in particle data space).
+    :param ymin: The starting y-coordinate (in particle data space).
+    :param pixcountx: The number of pixels in the output image in the x-direction.
+    :param pixcounty: The number of pixels in the output image in the y-direction.
+    :param int_samples: The number of sample points to take when approximating the 2D column kernel.
+    :return: The output image, in a 2-dimensional numpy array.
+    """
```

Comment:
Can you change to use Numpy style docstrings?

Style guide here: https://numpydoc.readthedocs.io/en/latest/format.html
Examples here: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html